src/rgw/rgw_sal_daos.h

```diff
@@ -267,7 +272,7 @@ class DaosBucket : public Bucket {
                               uint64_t timeout) override;
   virtual int purge_instance(const DoutPrefixProvider* dpp) override;
   virtual std::unique_ptr<Bucket> clone() override {
-    return std::make_unique<DaosBucket>(*this);
+    return std::unique_ptr<DaosBucket>(new DaosBucket(*this));
```

Comment:
This is what `make_unique` literally does, so no new changes here.

Reply:
I know that is the expectation, but there seems to be a subtle difference and the change does resolve a crash.  The copy constructor is also needed.